scipy/ndimage/tests/test_measurements.py

```diff
@@ -101,236 +116,259 @@ def test_basic(self):
                     [0, 0, 0, 1, 1, 0]]
+        expected = xp.asarray(expected)
         assert_array_almost_equal(out, expected)
-        assert_equal(n, 1)
+        assert n == 1
 
 
-def test_label13():
+def test_label13(xp):
     for type in types:
-        data = np.array([[1, 0, 1, 1, 1, 0, 1, 1, 1, 0, 1],
-                         [1, 1, 1, 0, 1, 1, 1, 0, 1, 1, 1],
-                         [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
-                         [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]],
-                        type)
+        dtype = getattr(xp, type)
+        data = xp.asarray([[1, 0, 1, 1, 1, 0, 1, 1, 1, 0, 1],
+                           [1, 1, 1, 0, 1, 1, 1, 0, 1, 1, 1],
+                           [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+                           [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]],
+                          dtype=dtype)
         out, n = ndimage.label(data)
         expected = [[1, 0, 1, 1, 1, 0, 1, 1, 1, 0, 1],
                     [1, 1, 1, 0, 1, 1, 1, 0, 1, 1, 1],
                     [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
                     [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]]
+        expected = xp.asarray(expected)
         assert_array_almost_equal(out, expected)
-        assert_equal(n, 1)
+        assert n == 1
 
 
-def test_label_output_typed():
-    data = np.ones([5])
+@skip_xp_backends(np_only=True, reasons=['output=dtype is numpy-specific'])
+def test_label_output_typed(xp):
+    data = xp.ones([5])
     for t in types:
-        output = np.zeros([5], dtype=t)
+        dtype = getattr(xp, t)
+        output = xp.zeros([5], dtype=dtype)
         n = ndimage.label(data, output=output)
-        assert_array_almost_equal(output, 1)
-        assert_equal(n, 1)
+        assert_array_almost_equal(output,
+                                  xp.ones(output.shape, dtype=output.dtype))
+        assert n == 1
 
 
-def test_label_output_dtype():
-    data = np.ones([5])
+@skip_xp_backends(np_only=True, reasons=['output=dtype is numpy-specific'])
+def test_label_output_dtype(xp):
+    data = xp.ones([5])
     for t in types:
-        output, n = ndimage.label(data, output=t)
-        assert_array_almost_equal(output, 1)
+        dtype = getattr(xp, t)
+        output, n = ndimage.label(data, output=dtype)
+        assert_array_almost_equal(output,
+                                  xp.ones(output.shape, dtype=output.dtype))
         assert output.dtype == t
 
 
-def test_label_output_wrong_size():
-    data = np.ones([5])
+def test_label_output_wrong_size(xp):
+    if is_jax(xp):
+        pytest.xfail("JAX does not raise")
+
+    data = xp.ones([5])
     for t in types:
-        output = np.zeros([10], t)
-        assert_raises((RuntimeError, ValueError),
+        dtype = getattr(xp, t)
+        output = xp.zeros([10], dtype=dtype)
+        # TypeError is from non-numpy arrays as output
+        assert_raises((ValueError, TypeError),
                       ndimage.label, data, output=output)
```

Comment:
this is totally fine, but for the future, I would personally leave tests for specific errors as `np_only` unless they are mentioned under `Raises` in the docs - they can always be changed in the future, it just seems a bit unnecessary to test that at this stage imo

Reply:
Agreed.